gendiff/core.py

```diff
@@ -11,36 +11,57 @@ def get_keys_groups(
 
     union = keys1_set | keys2_set
     common = keys1_set & keys2_set
-    missing = keys1_set - common
 
     return {
         "union": union,
         "common": common,
-        "missing": missing,
+        "missing": keys1_set - common,
+        "new": keys2_set - common,
     }
 
 
-def get_diff_str(dict1: Dict[str, Any], dict2: Dict[str, Any]) -> str:
+def get_diff_view(dict1: Dict[str, Any], dict2: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
let's update this function to store more information (original and changed values), so we could leave original dicts alone